tools/Az.Tools.Predictor/Az.Tools.Predictor/ParameterValuePredictor.cs

```diff
@@ -27,7 +24,15 @@ public ParameterValuePredictor()
             var fileInfo = new FileInfo(typeof(Settings).Assembly.Location);
             var directory = fileInfo.DirectoryName;
             var mappingFilePath = Path.Join(directory, "command_param_to_resource_map.json");
-            _command_param_to_resource_map = JsonSerializer.Deserialize<Dictionary<string, Dictionary<string, string>>>(File.ReadAllText(mappingFilePath), JsonUtilities.DefaultSerializerOptions);
+
+            try
+            {
+                _command_param_to_resource_map = JsonSerializer.Deserialize<Dictionary<string, Dictionary<string, string>>>(File.ReadAllText(mappingFilePath), JsonUtilities.DefaultSerializerOptions);
+            }
+            catch
+            {
+                // We don't want it to crash the module when the file doesn't exist or when it's mal-formatted.
```

Comment:
Could (or should) we add telemetry to report a crash?

Reply:
Good point. I add it.